server/service/src/sync/sync_on_central/mod.rs

```diff
@@ -180,3 +182,96 @@ pub async fn download_file(
         actix_files::NamedFile::open(&file_description.path).map_err(|e| Error::from_error(&e))?;
     Ok((named_file, file_description))
 }
+
+/// Accept a file from a remote open-mSupply Server
+/// This is the endpoint that the remote server will call to upload a file
+pub async fn upload_file(
+    settings: &Settings,
+    service_provider: &ServiceProvider,
+    file_id: String,
+    SyncUploadFileRequestV6 {
+        files,
+        sync_v5_settings,
+    }: SyncUploadFileRequestV6,
+) -> Result<(), SyncParsedErrorV6> {
+    use SyncParsedErrorV6 as Error;
+
+    log::info!("Receiving a file via sync : {}", file_id);
+
+    if !is_central_server() {
+        return Err(Error::NotACentralServer);
+    }
+    // Check credentials again mSupply central server
+    let _ = SyncApiV5::new(sync_v5_settings.into_inner())
+        .map_err(|e| Error::OtherServerError(format_error(&e)))?
+        .get_site_info()
+        .await
+        .map_err(Error::from)?;
+
+    let file_service = StaticFileService::new(&settings.server.base_dir)?;
+    let db_connection = service_provider
+        .connection()
+        .map_err(|e| Error::OtherServerError(format_error(&e)))?;
+
+    let repo = SyncFileReferenceRowRepository::new(&db_connection);
+    let mut sync_file_reference = repo
+        .find_one_by_id(&file_id)
+        .map_err(|e| Error::OtherServerError(format_error(&e)))?
+        .ok_or({
+            log::error!(
+                "Sync File Reference not found, can't upload until this is synced: {}",
+                file_id
+            );
+            Error::SyncFileNotFound
+        })?;
+
+    // for each uploaded file reserve a file in the static files directory, then copy the file from the temp file location
+    // Should only be 1 file, but we will loop through them all just in case we need to do some clean up..
+    for file in files {
+        let static_file_category = StaticFileCategory::SyncFile(
+            sync_file_reference.table_name.clone(),
+            sync_file_reference.record_id.clone(),
+        );
+        let sanitized_filename = file.file_name.map(sanitize_filename).unwrap_or_default();
+
+        let static_file = file_service.reserve_file(
+            &sanitized_filename,
+            &static_file_category,
+            Some(file_id.clone()),
+        )?;
+        let destination = Path::new(&static_file.path);
+
+        // Copy the file from the temp location to the final location
+        // TODO: Ideally these fs operations should be done in a separate such as using web::block (see handle_upload in static_files.rs)
```

Comment:
I had lifetime issues doing this, so figured we'd probably get away with this in the short term. Potentially to hang the webserver if there was some kind of disk issue but that probably suggests bigger issues...